i18n: German translation

HeyMyian opened the pull request:
**Summary:**
Complete German translation, created in Crowdin and tested in current novelWriter release (1.6.2).
My first ever pull request. Please let me know in case I missed anything.

**Reviewer's Checklist:**

* [x] The header of all files contain a reference to the repository license
* [x] The overall test coverage is increased or remains the same as before
* [x] All tests are passing
* [ ] All flake8 checks are passing and the style guide is followed
* [x] Documentation (as docstrings) is complete and understandable
* [x] Only files that have been actively changed are committed

vkbo commented:
Does not pass linting check: https://github.com/vkbo/novelWriter/runs/5823808465?check_suite_focus=true

The translation needs to be verified on Crowdin, but I have no problem with accepting the one here for now and update it before release if we need to change a few.

HeyMyian commented:
I fixed the missing space for the linting check and updated the `.ts` file from Crowdin. There were so many incorrect spellchecking and translation warnings on Crowdin that I automatically glanced over the ones that were valid.

I understand if you want to have more eyes on the Crowdin source to verify my translation.

vkbo commented:
Thanks! Now it passes.

After going through a bunch yesterday, I saw a few issues. There are a number of translations that have less information in the German than the English. While you can rephrase things for clarity between languages, the translations still need to have the same information content. For instance, in the project menu you did not include the German word for "project" in the translations. That needs to be there otherwise the menu is ambiguous. Not everything in it is about the projects.

There are other examples too. Some of the dialog box texts also have a less information than in the English text.

HeyMyian commented:
> For instance, in the project menu you did not include the German word for "project" in the translations. That needs to be there otherwise the menu is ambiguous. Not everything in it is about the projects.

Are you referring to the menu, like `Project > New`, `Project > Open`, `Document > Close`? I chose to leave out the words "Project" and "Document" in those because they already live in a menu that is called like that and I wanted to avoid being overly verbose. I guess I have overdone it with cutting words out.

I'll go back to Crowdin and make sure all information is there. Would you prefer I notify you before I create another pull request?

vkbo commented:
> Are you referring to the menu, like `Project > New`, `Project > Open`, `Document > Close`? I chose to leave out the words "Project" and "Document" in those because they already live in a menu that is called like that and I wanted to avoid being overly verbose. I guess I have overdone it with cutting words out.

Yes, those were some examples. As I said, they are named that way for a reason. If the verbosity is to be changed anywhere, that's a general discussion. I'm happy for any feedback on the GUI text and dialog boxes. I've made a lot of changes due to good suggestions from translators. However, they should be changed in the English first, and then translated to the various languages.

To clarify the point I made in the translation thread. It is ok to rephrase due to linguistic differences, and even so the phrases or words are short enough in some space-critical places. The latter is an exception, and only applies to a handful of strings. Generally, all translations should have the same content. Many places have detailed explanations to avoid misunderstandings. They may be verbose, but sometimes that is because users have asked for clarifications or suggested better texts.

Example:

English -> German
"File Changed on Disk" -> "Konflikt"
"Please select some text before calling replace quotes." -> "Bitte erst Text markieren."

These are not the same.

Crowdin sometimes complains about inconsistencies. But these are too complex for it to catch.

HeyMyian commented:
Understood. I'll revise the translation to fit the criteria.

vkbo commented:
Oh, no need to close the pull request. Just have a quick run-through again and update the .ts file. :)

I will also take some time to review the translations and approve them. I have gone through a few already. I'm just a bit busy this week as I'm at a conference.

It will still be a while until 1.7 is released, so there is no rush to complete this.

Thanks again for your effort!

vkbo commented:
Ok, I've approved about 75% now. One thing I noticed, and that Crowdin also noticed, is that you use two different words for "folder".

HeyMyian commented:
> folder

I used "Verzeichnis" for the file structure on disk (project folder and backup folder) and "Ordner" for the organisational tool within novelWriter (all the possible folders in the project tree).

I have a vague memory of "Verzeichnis" being the "more correct" translation for the folder structure of a file system, whereas "Ordner" is for a collection of documents (also on computers). My distinction might not make a lot of sense for people who don't know the project tree folders don't exist on disk. If you would want to select one consistent translation, I'd suggest "Ordner".

vkbo commented:
> Edit: Oh yeah, English has the same distinction: directory and folder.

Yup. "Folder" is the preferred in English here, since it is used to describe a collection of documents. So "ordner" would be the German equivalent, which makes total sense as a Norwegian too!

vkbo commented:
Ok, I'm back from vacation now. I'll see if I can have a look at the translations tomorrow so we can complete the merge.

HeyMyian commented:
I've been wondering if you decided on the folder (Ordner) vs. directory (Verzeichnis) distinction. I've seen you approved translations with either word, and I've also seen that in some cases I mistakenly hadn't deleted my alternate proposed (for my version: wrong) translations.

vkbo commented:
Yes, as I said "ordner" probably makes more sense, but it's up to you! I may have approved a few of either as it took a while before I noticed the difference. You can just propose a new translation to correct it if you know which ones to update.

HeyMyian commented:
I'll double-check all translations over the weekend and also look at the ones that were lacking previously. 👍

HeyMyian commented:
Current status is I cleaned up my translations on Crowdin and am testing it within novelWriter. I'm having a bit of trouble finding strings that I gave more than one translation for, or searching for multiple different words at the same time in Crowdin. Those would show me whatever I was thinking at the time of translation and help with consistency. I think when building the `.ts` or `.qm` file, only the most recent (or approved) translation is preserved.

I haven't yet resolved the "Ordner" <-> "Verzeichnis" thing for that reason, and now also threw in another possible translation to the mix, "Speicherort". That said, _imo_ the translation is fully functional except for the minor consistency issues. If you have any comments or would like something to be handled differently, I'm happy to adapt.

What was your time frame for the next release?

vkbo commented:
I have a milestone for each release, which include the features I intend to implement. The 1.7 Beta 1 list is here: https://github.com/vkbo/novelWriter/milestone/38

The less stuff on the list, the closer I am to the release. I may move some of the larger issues to 1.8 though, so the list is not set in stone. I've made some major changes under the hood that makes it easier to implement some of the planned features, so I may speed things up a bit. In any case, there will definitely be a Beta 1 and RC 1 release before the full 1.7 release. I never add any new features in RC releases, but adding translations are fine, so there is plenty of time.

vkbo commented:
Hi again. I've gone through the remaining translations needing proofreading and approved most of them. The ones that remain have a translation that is different than the original English, that is, they don't say the same thing, so they need to be updated.

vkbo commented:
I see there are two conflicts now with the main branch. They are simply due to the fact that I sorted the translators list alphabetically in about and credits, so they should be easy to fix.

Please let me know here when you've had the time to look through the remaining translations, and I will approve them. There should only be about 20 left. Then you can update the .ts file in this pull request and I will merge it,

Now that the beta release is out for 1.7, it will not be too long before I make the RC1 release and then the final release. I would very much like to include the German localisation in it!

HeyMyian commented:
I went through the translations that needed approval. I added new translations for some, although in some cases I personally prefer the older version. I'm not really married to any of them though, so have at it :)

The ones I did not change were either already obviously correct (`Export Options` > `Exportoptionen`) or I don't see how else they could be translated while maintaining natural German language. `Build Novel Project` is an example for those; I don't know what else this could be but `Projekt exportieren`. `Projekt bauen` (literal translation) is unnatural, and other options like `Projekt erstellen` or `Projekt kompilieren` aren't working either, imo.

About all the quotation marks:
E.g. `Double right-pointing angle quotation mark` > `Guillemet rechtszeigend`: that's just what that is, a Guillemet that points to the right (not closing or opening, because usage differs between text languages). The others, like `66 oben` I think are accepted forms of calling them in German. `66 oben` specifically is the closing quotation mark in German; wouldn't be able to translate that literally from `Left double quotation mark`.
As a note, perhaps the names for quotation marks should be separated from the GUI language -- as in, not denoting if a symbol is opening or closing) because there are some wild variations out there and I don't think we can expect people to write in the same language as their GUI language.

vkbo commented:
> The ones I did not change were either already obviously correct (`Export Options` > `Exportoptionen`) or I don't see how else they could be translated while maintaining natural German language.

That one is fine. I may have missed a couple.

> `Build Novel Project` is an example for those; I don't know what else this could be but `Projekt exportieren`. `Projekt bauen` (literal translation) is unnatural, and other options like `Projekt erstellen` or `Projekt kompilieren` aren't working either, imo.

My issue here is that it is _not_ an export. It is a build tool. How do you describe building/compiling code in German?

> About all the quotation marks: E.g. `Double right-pointing angle quotation mark` > `Guillemet rechtszeigend`: that's just what that is, a Guillemet that points to the right (not closing or opening, because usage differs between text languages). The others, like `66 oben` I think are accepted forms of calling them in German. `66 oben` specifically is the closing quotation mark in German; wouldn't be able to translate that literally from `Left double quotation mark`. As a note, perhaps the names for quotation marks should be separated from the GUI language -- as in, not denoting if a symbol is opening or closing) because there are some wild variations out there and I don't think we can expect people to write in the same language as their GUI language.

The labels are taken from the [Wikipedia article](https://en.wikipedia.org/wiki/Quotation_mark), which I think just takes them from the Unicode standard. My issue with "Guillemet rechtszeigend" is that it doesn't say "double" while the other ones do say "single". That's inconsistent. My issue with the others is that they are short. The English version is a full description.

There are also a few translations that change the meaning, which I cannot accept.

* "Title Formats for Novel Files" > "Überschriftenformate" : This is misleading. The title format is _only_ for Novel Files, and the label has to say that. Otherwise German speaking users may wonder why they don't work for note files.
* "Include in build" > "Exportieren" : This is not saying the same thing.
* "Populate Novel Folder" > "Kapitel und Szenen" : This is not saying the same thing.
* "Fill the project with a minimal set of items" > "Minimal notwendige Elemente hinzufügen" : This is simply wrong. There are no "necessary" files in a project except the Novel folder itself.
* "Fill the project with example files" > "Beispielprojekt erstellen" : Similar, but still not saying the same thing.
* "Show detailed options for filling the project" > "Benutzerdefinierte Struktur" : Also similar, but not saying the same thing.

As I've mentioned before, it is fine to reword things to fit differences in languages. But you cannot change the fundamental meaning of the labels. The main reason why you cannot do that is that if the GUI says different things in different languages, it becomes very difficult to give support to people who use a GUI in a different language and to figure out why they may misunderstand a feature or use it in an unintended way. It also makes the documentation inconsistent with the GUI.

As also mentioned before, the proper way to change the meaning of a label is to file an issue or suggest an improvement on the original label in the code, which will then propagate to all translations. That's how labels are changed.

HeyMyian commented:
I've been agonizing over the remaining translations.

**Quotation marks**
I changed the "straight" quotation marks, this is better. I fixed the identical label, that was an accident. The others I can't with a clear conscience change. "Guillemets", as that word, are double in German; "einfache Guillemets" are single. 66 oben, 99 unten, etc. describe which quotation mark it is in the most concise and precise way. "Double low-9 quotation mark" turns to "99 unten". It could be `Doppeltes Anführungszeichen 9 unten` but the only thing gained by translating them that way is letter count. I've added the translations anyway so you can choose, but I really don't see why.

**Title Format for Novel Files**
I leaned on the existing Norwegian and Spanish translations which also leave out "novel files". Fixed.

The strings about **Build**
I think "Roman zusammenbauen" or a variation of that could work. I suggested some in Crowdin.

**Writing Statistics**
There is already an approved identical string. "Schreibstatistiken" would be just about the clunkiest word I've ever read in the German language.

**The ones missing in the new project dialogue**
They are giving me a really hard time. I understand the problems with them as you described. I'll mull over them some more if I have the time before the new version.

Is there maybe another German speaking person here who has an idea?

vkbo commented:
> but the only thing gained by translating them that way is letter count

These are descriptive labels, so there is no need for brevity. In fact, being explicit is the entire point of them. I get that you are very resistant to this for some reason, but it needs to be consistent.

> I leaned on the existing Norwegian and Spanish translations which also leave out "novel files". Fixed.

Then that needs to be fixed too. Although there isn't a lot of room on this label. In any case, these will change in 1.8 as I'm redesigning the whole tool. It may not be that important.

> Is there maybe another German speaking person here who has an idea?

One of my old colleagues who helped planning the original version of novelWriter may have some input, @Number042.

The New Project Wizard will change a bit before the final release of 1.7, but the three options on that page will not. All I ask is that you fix the one that says something that is directly wrong.

vkbo commented:
About the New Project Wizard. There are a bunch of reasonable automated translation options there that may just need som grammatical tweaking. Why not use those as a starting point? They are complete translations it seems as well.

TannersAndRodents commented:
I would consider the following translations for the wizard:


`Fill the project with a minimal set of items` > `minimale Elemente zum Projekt hinzufügen` (clunky, but I can't think of a better translation)


`Fill the project with example files` > `Beispielelemente zum Projekt hinzufügen`


`Show detailed options for filling the project` > `erweiterte Optionen für das Hinzufügen von Elementen anzeigen`


About `Build Project`: `zusammenbauen` is weird because it is only used in a literal sense (e.g. with puzzle, computer components). It's always `exportieren` in different programs and I get that it's not an export, but that's what users will search for. My preferred "non-export" option would be `kompilieren`. However, this term is mainly used for programming, or (with negative connotation) piecing together other people's literary works. Alternatives are `zusammenstellen` or maybe something with `setzen` (which means typesetting).

vkbo commented:
Whatever the three Wizard options are, they must be complete sentences, not sentence fragments. They must also not change the meaning of what they say, as one of the proposed translations do.

As for the issue with "Export" vs "Build". The reason I want to keep them separate is that there are actual export options planned, which are different than building a manuscript. Maybe it would be easier if the English label too was more descriptive. The entire tool will be rebuilt in the next version, so I'll keep this in mind.

vkbo commented:
Proposal to get this finished:

* Make the three Wizard options complete sentences.
* Use the German equivalent for export for now until I redesign the build tool. I will try to find a better description in English. It is a bit clunky in Norwegian too to be honest.

Then, please upload the new ts file to this pull request, and I can merge it.

Due to the changes I'm making to internal functionality to accommodate new features, I will need to make a second beta release for testing. There will be a few new translations for the RC 1 release after that.

Again, thanks a lot for all your input and contributions!

vkbo commented:
Hi! I would like to have the German translation in the 2.0 release. It is probably a good time now to complete it as there will not be any more updates to translation files. What mostly remains before 2.0 is fixing any issues that arise during testing, and update the documentation.

vkbo commented:
I'm just going to merge this. I will pull the latest file from Crowdin at the 2.0 release if it's ready. It seems to be pretty close. I approved your latest translations.

HeyMyian commented:
I haven't checked the testing version of the big update, so I'm at a loss what the context for these last translation-less strings is. I've provided some alternatives for you to choose from.

Is there anything else now needed for the German translation?

As I recall there might still be some inconsistencies with what I named things? Especially all strings that refer to "build", which I have been told in the meantime is an acceptable word to use in German too in that context (and is needed as such now anyway because of the small space in the left toolbar).

---

> Include inactive files
https://crowdin.com/translate/novelwriter/1047/engb-de#130167

What constitutes an inactive file? Is that a new name for the "Included Flag"?

vkbo commented:
Hi,

I've updated screenshots for the things that are on the GUI, but not everything is. The only way for you to check is to run the latest release candidate really. However, you can always provide fixes once 2.0 is released and I will include it in the next patch. There are likely to be a few with such a big release.

With regards to "build" vs "export", I have made this change consistently in the documentation now. As mentioned before, there will be an actual export feature added soon that is different from the manuscript build, so I wanted to clear the way for it.

As for the inactive, yes the include/exclude flag has become active/inactive instead.
